src/main/java/edu/wpi/grip/ui/pipeline/input/NumberSpinnerInputSocketView.java

```diff
@@ -0,0 +1,66 @@
+package edu.wpi.grip.ui.pipeline.input;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.events.SocketChangedEvent;
+import javafx.beans.InvalidationListener;
+import javafx.scene.control.Spinner;
+import javafx.scene.control.SpinnerValueFactory;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * An {@link InputSocketView} that lets the user set the value of a number with a {@link Spinner}
+ */
+public class NumberSpinnerInputSocketView extends InputSocketView<Number> {
+
+    private final static Object[] DEFAULT_DOMAIN = new Object[]{Double.MIN_VALUE, Double.MAX_VALUE};
+
+    private final SpinnerValueFactory<Double> valueFactory;
+    private final InvalidationListener updateSocketFromSpinner;
+
+    /**
+     * @param socket An <code>InputSocket</code> with a domain containing two <code>Number</code>s (the min and max
+     *               slider values), or no domain at all.
+     */
+    public NumberSpinnerInputSocketView(EventBus eventBus, InputSocket<Number> socket) {
+        super(eventBus, socket);
+
+
+        Object[] domain = socket.getSocketHint().getDomain();
+        if (domain == null) {
+            domain = DEFAULT_DOMAIN;
+        }
+
+        checkArgument(domain.length == 2 && domain[0] instanceof Number && domain[1] instanceof Number,
+                "Spinners must have a domain with two numbers (min and max)");
+
+        final double min = ((Number) domain[0]).doubleValue();
+        final double max = ((Number) domain[1]).doubleValue();
+        final double initialValue = socket.getValue().doubleValue();
+
+        this.valueFactory = new SpinnerValueFactory.DoubleSpinnerValueFactory(min, max, initialValue);
+        this.updateSocketFromSpinner = o -> this.getSocket().setValue(this.valueFactory.getValue());
+        this.valueFactory.valueProperty().addListener(this.updateSocketFromSpinner);
+
+        final Spinner<Double> spinner = new Spinner<>(this.valueFactory);
+        spinner.setEditable(true);
+        spinner.disableProperty().bind(this.getHandle().connectedProperty());
```

Comment:
Because it's applied to a different control in each subclass, and there's no way for the parent class to get that control (adding a way to do that would probably be considered breaking encapsulation).  

Using `Property` objects to bind the state to each subclass seemed like an appropriate way to nicely wrap this sort of thing.